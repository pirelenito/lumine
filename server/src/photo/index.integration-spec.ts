/**
 * Stub id generation to get predictable results
 */
jest.mock('./generateId', () => {
  return (fullPath: string) => crypto.createHash('md4').update(fullPath).digest('hex').slice(0, 4)
})

import { join } from 'path'
import { readdirSync } from 'fs'
import rimraf from 'rimraf'
import crypto from 'crypto'

import { importPhoto } from '.'
import { exiftool } from 'exiftool-vendored'

const libraryBasePath = join(__dirname, '../../fixtures')
const cacheBasePath = join(__dirname, '../../tmp')
const config = { libraryBasePath, cacheBasePath, httpPort: 80 }

/**
 * Cleanup any cache we have stored
 */
beforeAll(() => rimraf.sync(cacheBasePath))

readdirSync(libraryBasePath)
  .filter((photoPath) => photoPath.match(/\.(arw|jpg|jpeg|mp4|avi|mov|mpg)$/i))
  .forEach((photoPath: string) => {
    it(`loads ${photoPath}`, async () => {
      const photo = await importPhoto(config, photoPath)
      expect(photo).toMatchSnapshot()
    }, 180000)
  })

/**
 * Stop the singleton instance of the exiftool, otherwise jest is unable to finish
 */
afterAll(() => {
  exiftool.end()
})
