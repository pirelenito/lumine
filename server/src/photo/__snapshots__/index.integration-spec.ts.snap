// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads canonS2is.jpg 1`] = `
Object {
  "contentHash": "13a4a500001674cc286200000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "Canon PowerShot S2 IS",
    "createdAt": 1187099113000,
    "gps": undefined,
  },
  "relativePath": "canonS2is.jpg",
}
`;

exports[`loads iphone3gs.JPG 1`] = `
Object {
  "contentHash": "12898800001674cc286200000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "iPhone 3GS",
    "createdAt": 1320514388000,
    "gps": Object {
      "altitude": 33.5587918,
      "latitude": -27.5788333333333,
      "longitude": -48.5223333333333,
    },
  },
  "relativePath": "iphone3gs.JPG",
}
`;

exports[`loads iphone5s.jpg 1`] = `
Object {
  "contentHash": "140a9100001674cc286200000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "iPhone 5s",
    "createdAt": 1391598352000,
    "gps": Object {
      "altitude": 61.07607608,
      "latitude": 48.8551166666667,
      "longitude": 2.36418888888889,
    },
  },
  "relativePath": "iphone5s.jpg",
}
`;

exports[`loads iphone6.jpg 1`] = `
Object {
  "contentHash": "f2764000001674cc286200000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "iPhone 6",
    "createdAt": 1533837990000,
    "gps": undefined,
  },
  "relativePath": "iphone6.jpg",
}
`;

exports[`loads nokia5.jpg 1`] = `
Object {
  "contentHash": "22b05500001674cc286200000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "Nokia 5",
    "createdAt": 1541677318000,
    "gps": undefined,
  },
  "relativePath": "nokia5.jpg",
}
`;

exports[`loads nokia7.mp4 1`] = `
Object {
  "contentHash": "11ef51200016b846409880000",
  "mediaType": "video",
  "metadata": Object {
    "cameraModel": undefined,
    "createdAt": 1561287086000,
    "gps": Object {
      "altitude": undefined,
      "latitude": 59.6547,
      "longitude": 17.9353,
    },
  },
  "relativePath": "nokia7.mp4",
}
`;

exports[`loads nokia620.jpg 1`] = `
Object {
  "contentHash": "15f05d00001674cc286200000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "Lumia 620",
    "createdAt": 1365192313000,
    "gps": Object {
      "altitude": 0.007704160247,
      "latitude": -27.6007616666667,
      "longitude": -48.4811608333333,
    },
  },
  "relativePath": "nokia620.jpg",
}
`;

exports[`loads nokia5130.jpg 1`] = `
Object {
  "contentHash": "483c8000001674cc286200000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "5130c-2",
    "createdAt": 1256217484000,
    "gps": undefined,
  },
  "relativePath": "nokia5130.jpg",
}
`;

exports[`loads nokia5200.jpg 1`] = `
Object {
  "contentHash": "bbc20000001676e8f30280000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": 5200,
    "createdAt": 1543748137000,
    "gps": undefined,
  },
  "relativePath": "nokia5200.jpg",
}
`;

exports[`loads s300d.jpg 1`] = `
Object {
  "contentHash": "a4c2d00000116f42673c80000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "u10D,S300D,u300D",
    "createdAt": 1198097069000,
    "gps": undefined,
  },
  "relativePath": "s300d.jpg",
}
`;

exports[`loads sonyA6000.arw 1`] = `
Object {
  "contentHash": "17f800000016c7b4fad380000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "ILCE-6000",
    "createdAt": 1534255754000,
    "gps": undefined,
  },
  "relativePath": "sonyA6000.arw",
}
`;

exports[`loads sonyNex3.arw 1`] = `
Object {
  "contentHash": "10380000001674cc286200000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "NEX-3N",
    "createdAt": 1391623582000,
    "gps": undefined,
  },
  "relativePath": "sonyNex3.arw",
}
`;

exports[`loads sonyNex3.jpg 1`] = `
Object {
  "contentHash": "58000000001674cc286200000",
  "mediaType": "photo",
  "metadata": Object {
    "cameraModel": "NEX-3N",
    "createdAt": 1391623582000,
    "gps": undefined,
  },
  "relativePath": "sonyNex3.jpg",
}
`;

exports[`loads sonya6000.MP4 1`] = `
Object {
  "contentHash": "70e33e000016985ba70300000",
  "mediaType": "video",
  "metadata": Object {
    "cameraModel": undefined,
    "createdAt": 1552784377000,
    "gps": undefined,
  },
  "relativePath": "sonya6000.MP4",
}
`;
